src/dsp_tools/utils/connection_live.py

```diff
@@ -198,14 +199,16 @@ def _handle_non_ok_responses(self, response: Response, retry_counter: int) -> No
         if should_retry(response):
             log_request_failure_and_sleep("Transient Error", retry_counter, exc_info=False)
             return None
+        msg = "Permanently unable to execute the network action. "
+        blame: Literal["server", "client"] = "server"
+        if original_str := regex.search(r'{"knora-api:error":"dsp\.errors\.(.*)","@context', str(response.content)):
+            msg += f"\n{' ' * 37}Original Message: {original_str.group(1)}\n"
+            blame = "client"
+        if original_str and original_str.group(1).startswith("OntologyConstraintException"):
+            blame = "client"
+        if blame == "client":
+            raise InvalidInputError(msg)
         else:
-            msg = "Permanently unable to execute the network action. "
-            if original_str := regex.search(r'{"knora-api:error":"dsp\.errors\.(.*)","@context', str(response.content)):
-                msg += f"\n{' ' * 37}Original Message: {original_str.group(1)}\n"
-                if original_str.group(1).startswith("OntologyConstraintException"):
-                    msg += f"See {WARNINGS_SAVEPATH} for more information."
-                    raise InvalidInputError(msg)
-            msg += f"See {WARNINGS_SAVEPATH} for more information."
```

Comment:
This is an attempt to make the code easier, so that I can do further work on it (concretely: https://linear.app/dasch/issue/DEV-4631/improvement-of-error-message-when-api-returns-notfoundexception).

Please tell me if you feel it was easier before...